apis/logging/v1/clusterlogging_types.go

```diff
@@ -178,34 +173,67 @@ type ElasticsearchSpec struct {
 
 // This is the struct that will contain information pertinent to Log and event collection
 type CollectionSpec struct {
-	// Specification of Log Collection for the cluster
-	Logs LogCollectionSpec `json:"logs,omitempty"`
-}
 
-type LogCollectionSpec struct {
 	// The type of Log Collection to configure
 	Type LogCollectionType `json:"type"`
 
+	// The resource requirements for Collector
+	//
+	// +nullable
+	// +optional
+	Resources *v1.ResourceRequirements `json:"resources"`
+
+	// Define which Nodes the Collector Pods are scheduled on.
+	//
+	// +nullable
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	Tolerations  []v1.Toleration   `json:"tolerations,omitempty"`
+
 	// Specification of the Fluentd Log Collection component
-	FluentdSpec `json:"fluentd,omitempty"`
+	//
+	// +nullable
+	// +optional
+	FluentdSpec *FluentdSpec `json:"fluentd,omitempty"`
+
+	// Specification of the Vector Log Collection component
+	//
+	// +nullable
+	// +optional
+	VectorSpec *VectorSpec `json:"vector,omitempty"`
 }
 
 type EventCollectionSpec struct {
 	Type EventCollectionType `json:"type"`
 }
 
 type FluentdSpec struct {
-	// The resource requirements for Fluentd
+	// Specification for Forwarder component for the cluster
 	//
+	// +optional
 	// +nullable
+	Tuning *FluentdTuningSpec `json:"tuning,omitempty"`
+}
+
+type VectorSpec struct {
+	//DataDir is the directory used for persisting Vector state
 	// +optional
-	Resources *v1.ResourceRequirements `json:"resources"`
+	DataDir string `json:"data_dir"`
 
-	// Define which Nodes the Pods are scheduled on.
-	//
-	// +nullable
-	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
```

Comment:
Why do we not need node selectors & tolerations for Vector? It's great if we don't - just curious why.